### ISSUE TITLE
Session timeout behavior modification

### ISSUE THREAD

**Issue by dspruell** [MISP/MISP#1287]
### Work environment

| Questions | Answers |
| --- | --- |
| Type of issue | Bug |
| OS version (server) | Ubuntu 14.04 |
| OS version (client) | OS X 10.10 |
| PHP version | PHP 5.5.9-1ubuntu4.17 |
| MISP version / git hash | 2.4.47 c26c0522be8338c88703e994fc962dad5b60e7ed |
| Browser | Firefox 46.0.1 |
### Expected behavior
1. Session timeouts should function like an idle timeout, clearing the session after interactions with site have not occurred for a defined amount of time.
2. There should be a user-modifiable configuration option for session timeout with a reasonable default set.
### Actual behavior

Session timeouts occur in the middle of active use of MISP. Users may be in the middle of data entry or actively browsing the interface of a site, and the session expiration appears to hit unrelated to inactivity, logging the user out and disrupting activity that may have been in process such as data entry.
### Steps to reproduce the behavior
1. Log in to MISP
2. Use site for extended period of time, navigating around to different views.
3. MISP login session will expire and return user to login screen after what appears to be static timeout, not idle timeout.

**Comment by RichieB2B** [comment]
I have experieced this on occasion as well. But certainly not all of the time. I fear this might be hard to reproduce.

**Comment by dspruell** [comment]
Yes, it's at odd enough intervals that it's difficult to describe when/what causes it from usage standpoint. In my recent occurrence moments, ago, I was working in an event (adding attributes) and clicked the "+" to add a new event and the dialog that appeared contained the login page, so a reload of the event page resulted in redirection to the login page.

**Comment by pinowudi** [comment]
Problem appears to be with a static 1-hour cookie expiry set in the session key at login that is never updated.
